client/network/src/block_request_handler.rs

```diff
@@ -85,6 +85,7 @@ struct SeenRequestsKey<B: BlockT> {
 	support_multiple_justifications: bool,
 }
 
+#[allow(clippy::derive_hash_xor_eq)]
```

Comment:
And here directly happened, what will happen in the future with clippy.

You add an `allow` without checking the code. `support_multiple_justifications` isn't hashed here.

Reply:
I did spot that. My understanding is that `support_multiple_justifications` would lead to the same hash but that's only potentially a perf issue. If it was the other way around and `support_multiple_justifications` was checked in the hash but not used in the eq then that would break hash maps.
Happy to add it into the hash - can't hurt..

Reply:
How should that lead to the same hash?

Reply:
At the moment if you create two SeenRequestsKeys with all the fields the same except for `support_multiple_justifications` then the current hash implementation will generate the same hash (because it doesn't take that field into account). Thus you'd get a hash collision and slight perf degradation on a hashmap retrial (but then again if having all the same fields except support_multiple_justifications isn't a realistic possibility then there's no perf degregation).

The critical part is that for objects A, B stored in a hashmap, if A == B then hash(A) must equal hash(B) and I believe the current definition does maintain that property.

Reply:
The current implementation allows to be `A != B` and `hash(A) == hash(B)`

Reply:
We are speaking here about hashing a `bool` that probably leads to no perf degradation.

Reply:
Have included the bool in the hash.

`A != B` and `hash(A) == hash(B)` is ok (it's a hash collision which is not unusual and hashmaps handle it fine.).
`A == B` and `hash(A) != hash(B)` is not ok.

(Incidentally have asked the formal methods WG if its possible to formally verify this kind of thing.)